src/components/discovery.ts

```diff
@@ -74,11 +73,17 @@ export class ClusterDuckTypeTableFactory implements ComponentFactory<any> {
 
     table.emptyContent = `There are no ${spec.names.plural}!`;
     // TODO: this should be handled externally
-    const duckVersions = versions.reduce((acc: ResourceMeta[], cur) => acc.concat(status.ducks[cur.name]), [])
-    const latestVersions = latestDuckTypeVersions(duckVersions)
-    latestVersions.sort((a, b) => a.kind.localeCompare(b.kind))
+    const duckMetas = versions.reduce((acc: ResourceMeta[], cur) => acc.concat(status.ducks[cur.name]), [])
+    const stored = StoredVersions(duckMetas)
+
+    const storedMetas = stored.reduce((acc: ResourceMeta[], cur) => {
+      const storedMeta = duckMetas.find(meta => meta.kind === cur[0] && meta.apiVersion.endsWith(cur[1].name))
+      return storedMeta ? acc.concat(storedMeta) : acc
+    }, [])
+    // const latestVersions = latestDuckTypeVersions(duckVersions)
```

Comment:
What this comment for?
Also looks like this was the only reference to latestDuckTypeVersions I think